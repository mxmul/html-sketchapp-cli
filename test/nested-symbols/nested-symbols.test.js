const { promisify } = require('es6-promisify');
const rimrafAsync = promisify(require('rimraf'));
const path = require('path');
const { exec } = require('child-process-promise');
const dirContentsToObject = require('../_utils/dirContentsToObject');

const distPath = path.join(__dirname, 'dist');

beforeEach(() => rimrafAsync(distPath));

test('generates symbolInstance nodes', async () => {
  await exec(
    [
      'node ../../bin/cli',
      '--puppeteer-args="--no-sandbox --disable-setuid-sandbox"',
      '--out-dir dist',
      '--file index.html'
    ].join(' '),
    { cwd: __dirname }
  );

  const output = await dirContentsToObject(distPath);
  expect(output).toMatchSnapshot();
});

test('fails when referencing unknown Symbol', (done) => {
  exec(
    [
      'node ../../bin/cli',
      '--puppeteer-args="--no-sandbox --disable-setuid-sandbox"',
      '--out-dir dist',
      '--file unknown-symbol.html'
    ].join(' '),
    { cwd: __dirname }
  ).catch(err => {
      expect(err.stderr).toMatch('Unknown symbol master: Some Symbol');
      done();
  });
});
