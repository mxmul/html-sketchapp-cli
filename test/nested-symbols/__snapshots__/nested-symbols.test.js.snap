// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generates symbolInstance nodes 1`] = `
Object {
  "document.asketch.json": Object {
    "_class": "document",
    "assets": Object {
      "_class": "assetCollection",
      "colors": Array [],
    },
    "currentPageIndex": 0,
    "do_objectID": "__GUID__",
    "enableLayerInteraction": true,
    "enableSliceInteraction": true,
    "foreignSymbols": Array [],
    "layerStyles": Object {
      "_class": "sharedStyleContainer",
      "objects": Array [],
    },
    "layerSymbols": Object {
      "_class": "symbolContainer",
      "objects": Array [],
    },
    "layerTextStyles": Object {
      "_class": "sharedTextStyleContainer",
      "objects": Array [],
    },
    "pages": Array [],
  },
  "document.page.json": Object {
    "_class": "page",
    "do_objectID": "__GUID__",
    "exportOptions": Object {
      "_class": "exportOptions",
      "exportFormats": Array [],
      "includedLayerIds": Array [],
      "layerOptions": 0,
      "shouldTrim": false,
    },
    "frame": Object {
      "_class": "rect",
      "constrainProportions": false,
      "height": 784,
      "width": 72,
      "x": 0,
      "y": 0,
    },
    "hasClickThrough": true,
    "horizontalRulerData": Object {
      "_class": "rulerData",
      "base": 0,
      "guides": Array [],
    },
    "includeInCloudUpload": true,
    "isFlippedHorizontal": false,
    "isFlippedVertical": false,
    "isLocked": false,
    "isVisible": true,
    "layerListExpandedType": 0,
    "layers": Array [
      Object {
        "_class": "symbolMaster",
        "backgroundColor": Object {
          "_class": "color",
          "alpha": 1,
          "blue": 1,
          "green": 1,
          "red": 1,
        },
        "changeIdentifier": 0,
        "do_objectID": "__GUID__",
        "exportOptions": Object {
          "_class": "exportOptions",
          "exportFormats": Array [],
          "includedLayerIds": Array [],
          "layerOptions": 0,
          "shouldTrim": false,
        },
        "frame": Object {
          "_class": "rect",
          "constrainProportions": false,
          "height": 17,
          "width": 11.546875,
          "x": 8,
          "y": 8,
        },
        "hasBackgroundColor": false,
        "hasClickThrough": true,
        "horizontalRulerData": Object {
          "_class": "rulerData",
          "base": 0,
          "guides": Array [],
        },
        "includeBackgroundColorInExport": true,
        "includeBackgroundColorInInstance": false,
        "includeInCloudUpload": true,
        "isFlippedHorizontal": false,
        "isFlippedVertical": false,
        "isLocked": false,
        "isVisible": true,
        "layerListExpandedType": 0,
        "layers": Array [
          Object {
            "_class": "text",
            "automaticallyDrawOnUnderlyingPath": false,
            "do_objectID": "__GUID__",
            "dontSynchroniseWithSymbol": false,
            "exportOptions": Object {
              "_class": "exportOptions",
              "exportFormats": Array [],
              "includedLayerIds": Array [],
              "layerOptions": 0,
              "shouldTrim": false,
            },
            "frame": Object {
              "_class": "rect",
              "constrainProportions": false,
              "height": 17,
              "width": 11.546875,
              "x": 0,
              "y": 0,
            },
            "glyphBounds": "",
            "heightIsClipped": false,
            "isFlippedHorizontal": false,
            "isFlippedVertical": false,
            "isLocked": false,
            "isVisible": true,
            "layerListExpandedType": 0,
            "layers": Array [],
            "lineSpacingBehaviour": 2,
            "name": "A",
            "nameIsFixed": false,
            "resizingConstraint": 47,
            "resizingType": 0,
            "rotation": 0,
            "shouldBreakMaskChain": false,
            "style": Object {
              "color": "rgb(0, 0, 0)",
              "fontFamily": "Times New Roman",
              "fontSize": 16,
              "fontWeight": 400,
              "textAlign": "start",
              "textDecoration": "none",
              "textTransform": "none",
            },
            "text": "A",
            "textBehaviour": 0,
          },
        ],
        "name": "Item/A",
        "nameIsFixed": false,
        "resizesContent": false,
        "resizingConstraint": 63,
        "resizingType": 0,
        "rotation": 0,
        "shouldBreakMaskChain": false,
        "style": Object {
          "_class": "style",
          "endDecorationType": 0,
          "miterLimit": 10,
          "startDecorationType": 0,
        },
        "symbolID": "__GUID__",
        "verticalRulerData": Object {
          "_class": "rulerData",
          "base": 0,
          "guides": Array [],
        },
      },
      Object {
        "_class": "symbolMaster",
        "backgroundColor": Object {
          "_class": "color",
          "alpha": 1,
          "blue": 1,
          "green": 1,
          "red": 1,
        },
        "changeIdentifier": 0,
        "do_objectID": "__GUID__",
        "exportOptions": Object {
          "_class": "exportOptions",
          "exportFormats": Array [],
          "includedLayerIds": Array [],
          "layerOptions": 0,
          "shouldTrim": false,
        },
        "frame": Object {
          "_class": "rect",
          "constrainProportions": false,
          "height": 17,
          "width": 10.671875,
          "x": 8,
          "y": 26,
        },
        "hasBackgroundColor": false,
        "hasClickThrough": true,
        "horizontalRulerData": Object {
          "_class": "rulerData",
          "base": 0,
          "guides": Array [],
        },
        "includeBackgroundColorInExport": true,
        "includeBackgroundColorInInstance": false,
        "includeInCloudUpload": true,
        "isFlippedHorizontal": false,
        "isFlippedVertical": false,
        "isLocked": false,
        "isVisible": true,
        "layerListExpandedType": 0,
        "layers": Array [
          Object {
            "_class": "text",
            "automaticallyDrawOnUnderlyingPath": false,
            "do_objectID": "__GUID__",
            "dontSynchroniseWithSymbol": false,
            "exportOptions": Object {
              "_class": "exportOptions",
              "exportFormats": Array [],
              "includedLayerIds": Array [],
              "layerOptions": 0,
              "shouldTrim": false,
            },
            "frame": Object {
              "_class": "rect",
              "constrainProportions": false,
              "height": 17,
              "width": 10.671875,
              "x": 0,
              "y": 0,
            },
            "glyphBounds": "",
            "heightIsClipped": false,
            "isFlippedHorizontal": false,
            "isFlippedVertical": false,
            "isLocked": false,
            "isVisible": true,
            "layerListExpandedType": 0,
            "layers": Array [],
            "lineSpacingBehaviour": 2,
            "name": "B",
            "nameIsFixed": false,
            "resizingConstraint": 47,
            "resizingType": 0,
            "rotation": 0,
            "shouldBreakMaskChain": false,
            "style": Object {
              "color": "rgb(0, 0, 0)",
              "fontFamily": "Times New Roman",
              "fontSize": 16,
              "fontWeight": 400,
              "textAlign": "start",
              "textDecoration": "none",
              "textTransform": "none",
            },
            "text": "B",
            "textBehaviour": 0,
          },
        ],
        "name": "Item/B",
        "nameIsFixed": false,
        "resizesContent": false,
        "resizingConstraint": 63,
        "resizingType": 0,
        "rotation": 0,
        "shouldBreakMaskChain": false,
        "style": Object {
          "_class": "style",
          "endDecorationType": 0,
          "miterLimit": 10,
          "startDecorationType": 0,
        },
        "symbolID": "__GUID__",
        "verticalRulerData": Object {
          "_class": "rulerData",
          "base": 0,
          "guides": Array [],
        },
      },
      Object {
        "_class": "symbolMaster",
        "backgroundColor": Object {
          "_class": "color",
          "alpha": 1,
          "blue": 1,
          "green": 1,
          "red": 1,
        },
        "changeIdentifier": 0,
        "do_objectID": "__GUID__",
        "exportOptions": Object {
          "_class": "exportOptions",
          "exportFormats": Array [],
          "includedLayerIds": Array [],
          "layerOptions": 0,
          "shouldTrim": false,
        },
        "frame": Object {
          "_class": "rect",
          "constrainProportions": false,
          "height": 36,
          "width": 1008,
          "x": 8,
          "y": 44,
        },
        "hasBackgroundColor": false,
        "hasClickThrough": true,
        "horizontalRulerData": Object {
          "_class": "rulerData",
          "base": 0,
          "guides": Array [],
        },
        "includeBackgroundColorInExport": true,
        "includeBackgroundColorInInstance": false,
        "includeInCloudUpload": true,
        "isFlippedHorizontal": false,
        "isFlippedVertical": false,
        "isLocked": false,
        "isVisible": true,
        "layerListExpandedType": 0,
        "layers": Array [
          Object {
            "_class": "symbolInstance",
            "do_objectID": "__GUID__",
            "exportOptions": Object {
              "_class": "exportOptions",
              "exportFormats": Array [],
              "includedLayerIds": Array [],
              "layerOptions": 0,
              "shouldTrim": false,
            },
            "frame": Object {
              "_class": "rect",
              "constrainProportions": false,
              "height": 18,
              "width": 1008,
              "x": 0,
              "y": 0,
            },
            "isFlippedHorizontal": false,
            "isFlippedVertical": false,
            "isLocked": false,
            "isVisible": true,
            "layerListExpandedType": 0,
            "layers": Array [],
            "name": "Item/A",
            "nameIsFixed": false,
            "resizingConstraint": 63,
            "resizingType": 0,
            "rotation": 0,
            "shouldBreakMaskChain": false,
            "style": Object {
              "_class": "style",
              "endDecorationType": 0,
              "miterLimit": 10,
              "startDecorationType": 0,
            },
            "symbolID": "__GUID__",
          },
          Object {
            "_class": "symbolInstance",
            "do_objectID": "__GUID__",
            "exportOptions": Object {
              "_class": "exportOptions",
              "exportFormats": Array [],
              "includedLayerIds": Array [],
              "layerOptions": 0,
              "shouldTrim": false,
            },
            "frame": Object {
              "_class": "rect",
              "constrainProportions": false,
              "height": 18,
              "width": 1008,
              "x": 0,
              "y": 18,
            },
            "isFlippedHorizontal": false,
            "isFlippedVertical": false,
            "isLocked": false,
            "isVisible": true,
            "layerListExpandedType": 0,
            "layers": Array [],
            "name": "Item/B",
            "nameIsFixed": false,
            "resizingConstraint": 63,
            "resizingType": 0,
            "rotation": 0,
            "shouldBreakMaskChain": false,
            "style": Object {
              "_class": "style",
              "endDecorationType": 0,
              "miterLimit": 10,
              "startDecorationType": 0,
            },
            "symbolID": "__GUID__",
          },
        ],
        "name": "Container",
        "nameIsFixed": false,
        "resizesContent": false,
        "resizingConstraint": 63,
        "resizingType": 0,
        "rotation": 0,
        "shouldBreakMaskChain": false,
        "style": Object {
          "_class": "style",
          "endDecorationType": 0,
          "miterLimit": 10,
          "startDecorationType": 0,
        },
        "symbolID": "__GUID__",
        "verticalRulerData": Object {
          "_class": "rulerData",
          "base": 0,
          "guides": Array [],
        },
      },
    ],
    "name": "html-sketchapp symbols",
    "nameIsFixed": false,
    "resizingConstraint": 63,
    "resizingType": 0,
    "rotation": 0,
    "shouldBreakMaskChain": false,
    "style": Object {
      "_class": "style",
      "endDecorationType": 0,
      "miterLimit": 10,
      "startDecorationType": 0,
    },
    "verticalRulerData": Object {
      "_class": "rulerData",
      "base": 0,
      "guides": Array [],
    },
  },
}
`;
